azkaban-common/src/main/c/execute-as-user.c

```diff
@@ -39,18 +39,23 @@ int INVALID_INPUT = 30;
 
 /*
  *  Change the real and effective user and group from super user to the specified user
- *  
+ *
  *  Adopted from:
  *  ./hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c
- *  
+ *
  */
 
-int change_user(uid_t user, gid_t group) {
+int change_user(char *username, uid_t user, gid_t group) {
```

Comment:
Can we double check?

for `initgroups()` I saw the signature. But `get*uid()` returns `uid`s which are probably not `char*`. Did we test this?
https://docs.oracle.com/cd/E36784_01/html/E36872/geteuid-2.html

The `getpwnam()` function for example translates `uid` to `username`
http://man7.org/linux/man-pages/man3/getpwnam.3.html